lib/src/rules/library_private_types_in_public_api.dart

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Avoid using private types in public APIs.';
+
+const _details = r'''
+
+**AVOID** using private types in public APIs.
```

Comment:
Since this is a fairly specific (and limited) implementation, maybe a few words describing what this will (and won't) check?

Reply:
Done.